Mod2/Lessons/Week3/SQLAggregates.md

```diff
@@ -0,0 +1,158 @@
+
+There are many [SQL Aggregate](https://www.sqlservertutorial.net/sql-server-aggregate-functions/) functions.  Today we are going to cover a few of the most commonly used.
+
+### Count
+
+`COUNT()` gives us the number of records in a particular column; or, the number of records in a table.
+
+```sql
+-- To get the number of records in a particular column
+SELECT COUNT(column_name) FROM table_name;
+
+-- To get the number of records in a table
+SELECT COUNT(*) FROM table_name;
+```
+
+> With a partner, use COUNT() to answer the following questions:
+> 1. How many artists exist?
+> 2. How many songs exist?
+> 3. How many songs have an associated artist?
+> 4. 🌶️Spicy! How many artists are from 'New York'?
+> 5. 🌶️Spicy! How many different song lengths are represented?
+> 
+> Be ready to share out!
+
+### Sum
+
+`SUM()` adds all the values in a particular column and returns the total.
+
+> With a partner, use SUM() to answer the following questions:
+> 1. What is the total play_count of all songs?
+> 2. What is the total play_count of all songs with artist_id `2`?
+> 3. 🌶️ Spicy! What is the total play_count of all songs with a play_count less than 25?
+> 
+> Be ready to share out!
+
+
+### Average
+
+`AVERAGE()`  returns the mean (average) of all the numbers in a column.
+
+> With a partner, use AVERAGE() to answer the following questions:
+> 1. What is the average length of all songs?
+> 2. What is the average length of all songs with a play_count greater than 25?
+> 3. 🌶️ Spicy! What is the average play_count of songs for each artist?
+> 
+> Be ready to share out!
+
+## Grouping
+
+Grouping allows us to arrange data into sub-sections (groups) so that we can answer more specific questions.  In some of the exercises above, we needed to use a `WHERE` clause to drill down to the specific data we were looking for; in one question, we even needed to use multiple SQL statements!  Grouping allows us to answer questions about sub-sections of our data in one SQL statement.
+
+Using `GROUP BY` is a bit more complicated than only using Aggregate functions. As you build your query, you need to determine what to do (or not do) with every column in your table.
+
+** Instructor Note ** Instead of watching the video below, you could walk through an example in real-time.
+
+Watch [this short video](https://www.loom.com/share/4ce05601a3654b4c99ae2ecb40560fab) that walks through building a grouped aggregate statement on the `artists` table.  In the video, the narrator has one additional column to show that you _can_ group and include multiple aggregates if you want to or need to.  If you would like to try out a version of the SQL statement from the video, use the script below:
+
+```sql
+SELECT hometown, COUNT(name) FROM artists
```

Comment:
I think would be useful for the instructor at least to demonstrate the full query from the video

SELECT COUNT(name), COUNT(years_active) FROM artists GROUP BY hometown;

So students can think through the confusion of having two counts without aliases.

Reply:
Ooh! Good call